docs/reference/upgrade/rolling_upgrade.asciidoc

```diff
@@ -7,19 +7,22 @@ a time so upgrading does not interrupt service. Running multiple versions of
 not supported, as shards cannot be replicated from upgraded nodes to nodes
 running the older version.
 
-We strongly recommend that when you upgrade you divide your cluster's nodes
-into the following two groups and upgrade the groups in this order:
+When performing a rolling upgrade:
 
-. Nodes that are not <<master-node,master-eligible>>. You can retrieve a list
+. Upgrade nodes that are not <<master-node,master-eligible>>. You can retrieve a list
 of these nodes with `GET /_nodes/_all,master:false` or by finding all the nodes
 configured with `node.master: false`.
 
-. Master-eligible nodes, which are the remaining nodes. You can retrieve a list
-of these nodes with `GET /_nodes/master:true`.
+. Upgrade nodes tier-by-tier, starting with the frozen tier.
+Complete the upgrade for all nodes in each data tier before moving to the next.
+Upgrade the frozen tier, then the cold tier, then the warm tier, and upgrade
+the hot tier last. This ensures {ilm-init} can continue to move data through
+the tiers during the upgrade.
```

Comment:
You (possibly deliberately) omitted this sentence, but I think it's helpful:

```suggestion
the tiers during the upgrade. You can get the list of nodes in a specific tier
with a `GET /_nodes` request, for example: `GET /_nodes/data_frozen:true/_none`.
```

Reply:
Yes. I wasn't entirely sure cross major versions we've had any changes to that endpoint (filtering by tier like that). So I left that out.

Reply:
somehow couldn't find it in the doc here: https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-info.html

Reply:
This should work in 7.17 indeed. There's a link to the relevant page here:

> All the nodes selective options are explained [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster.html#cluster-nodes).

However, the linked page doesn't actually mention filtering by role. We should fix this too.